joins.sql

```diff
@@ -0,0 +1,123 @@
+
+SELECT comments.*, posts.title as "post_title", posts.url as "post_url", comments.body as "comment_body"
+FROM comments
+INNER JOIN posts on posts.id = comments.post_id
+WHERE posts.created_at < '2015-01-01';
+
+
+
+-- 1. Create a query to get the all rows in the `comments` table, showing post title (aliased as post_title), post url (ailased as post_url), and the comment body (aliased as comment_body) where the post was created after January 1, 2015
+
+SELECT comments.*, posts.title as "post_title", posts.url as "post_url", comments.body as "comment_body"
+FROM comments
+INNER JOIN posts on posts.id = comments.post_id
+WHERE posts.created_at > '2015-01-01';
+
+
+
+-- 1. Create a query to get the all rows in the `comments` table, showing post title (aliased as post_title), post url (ailased as post_url), and the comment body (aliased as comment_body) where the comment body contains the word 'USB'
+
+SELECT comments.*, posts.title as "post_title", posts.url as "post_url", comments.body as "comment_body"
+FROM comments
+INNER JOIN posts on posts.id = comments.post_id
+WHERE comments.body LIKE '%USB%';
+
+
+
+-- 1. Create a query to get the post title (aliased as post_title), first name of the author of the post, last name of the author of the post, and comment body (aliased to comment_body), where the comment body contains the word 'matrix' ( should have 855 results )
+
+SELECT posts.title as "post_title", users.first_name, users.last_name, comments.body as "comment_body"
+FROM comments
+INNER JOIN posts on posts.id = comments.post_id
+INNER JOIN users on comments.user_id = users.id
+WHERE comments.body LIKE '%matrix%';
+
+
+
+-- 1. Create a query to get the first name of the author of the comment, last name of the author of the comment, and comment body (aliased to comment_body), where the comment body contains the word 'SSL' and the post content contains the word 'dolorum' ( should have 102 results )
+
+SELECT users.first_name, users.last_name, comments.body as "comment_body"
+FROM comments
+INNER JOIN users on comments.user_id = users.id
+INNER JOIN posts on posts.id = comments.post_id
+WHERE comments.body LIKE '%SSL%' and posts.content LIKE '%dolorum%';
+
+
+-- 1. Create a query to get the first name of the author of the post (aliased to post_author_first_name), last name of the author of the post (aliased to post_author_last_name), the post title (aliased to post_title), username of the author of the comment (aliased to comment_author_username), and comment body (aliased to comment_body), 
+-- where the comment body contains the word 'SSL' or 'firewall' and the post content contains the word 'nemo' ( should have 218 results )
+
+SELECT users.first_name as "post_author_first_name", users.last_name as "post_author_last_name", posts.title as "post_title", users.username as "comment_author_username", comments.body as "comment_body"
```

Comment:
This is actually getting the first and last name of the commenter, not the post author because you've only joined the users table to this query once.